service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -580,4 +592,73 @@ public void deleteSiteImage(JWTData userData, int imageId) {
+      filterCondition = filterCondition.and(ADOPTED_SITES.DATE_ADOPTED.ge(filterSitesRequest.getAdoptedStart()));
+    if (filterSitesRequest.getAdoptedEnd() != null)
+      filterCondition = filterCondition.and(ADOPTED_SITES.DATE_ADOPTED.le(filterSitesRequest.getAdoptedEnd()));
+    if (filterSitesRequest.getNeighborhoodIds() != null)
+      filterCondition = filterCondition.and(SITES.NEIGHBORHOOD_ID.in(filterSitesRequest.getNeighborhoodIds()));
+
+    Condition stewardshipCondition = noCondition();
+    if (filterSitesRequest.getLastActivityStart() != null)
+      stewardshipCondition = stewardshipCondition.and(max(STEWARDSHIP.PERFORMED_ON).ge(filterSitesRequest.getLastActivityStart()));
+    if (filterSitesRequest.getLastActivityEnd() != null)
+      stewardshipCondition = stewardshipCondition.and(max(STEWARDSHIP.PERFORMED_ON).le(filterSitesRequest.getLastActivityEnd()));
+
+    Table<org.jooq.Record3<Integer, Integer, Integer>> activityCounts = db
+        .select(count().as(ACTIVITY_COUNT_COLUMN), STEWARDSHIP.SITE_ID, STEWARDSHIP.USER_ID)
+        .from(STEWARDSHIP)
+        .groupBy(STEWARDSHIP.SITE_ID, STEWARDSHIP.USER_ID).asTable("activityCounts");
+
+    Result<org.jooq.Record12<Integer, String, Integer, Integer, Date, Date, Timestamp, String, String, String, String, Integer>>
+        records = db
+        .select(SITES.ID, SITES.ADDRESS, SITES.NEIGHBORHOOD_ID, ADOPTED_SITES.USER_ID, ADOPTED_SITES.DATE_ADOPTED,
+            max(STEWARDSHIP.PERFORMED_ON).as(STEWARDSHIP.PERFORMED_ON),
+            max(SITE_ENTRIES.UPDATED_AT).as(SITE_ENTRIES.UPDATED_AT.getName()),
+            SITE_ENTRIES.SPECIES, USERS.FIRST_NAME, USERS.LAST_NAME, USERS.EMAIL, coalesce(activityCounts.field(ACTIVITY_COUNT_COLUMN, Integer.class), 0).as(ACTIVITY_COUNT_COLUMN))
+        .from(ADOPTED_SITES)
+        .join(SITES).on(ADOPTED_SITES.SITE_ID.eq(SITES.ID))
+        .join(USERS).on(ADOPTED_SITES.USER_ID.eq(USERS.ID))
+        .leftJoin(STEWARDSHIP).on(ADOPTED_SITES.SITE_ID.eq(STEWARDSHIP.SITE_ID))
+        .join(SITE_ENTRIES).on(ADOPTED_SITES.SITE_ID.eq(SITE_ENTRIES.SITE_ID))
+        .leftJoin(activityCounts).on(
+            ADOPTED_SITES.SITE_ID.eq(activityCounts.field(STEWARDSHIP.SITE_ID))
+                .and(ADOPTED_SITES.USER_ID.eq(activityCounts.field(STEWARDSHIP.USER_ID))))
+        .where(filterCondition)
+        .groupBy(SITES.ID, SITES.ADDRESS, SITES.NEIGHBORHOOD_ID, ADOPTED_SITES.USER_ID, ADOPTED_SITES.DATE_ADOPTED, SITE_ENTRIES.SPECIES, USERS.FIRST_NAME, USERS.LAST_NAME, USERS.EMAIL, activityCounts.field(ACTIVITY_COUNT_COLUMN))
+        .having(stewardshipCondition)
+        .fetch();
+
+    return records.stream().map(rec -> {
+      String adopterName = rec.get(USERS.FIRST_NAME) + ' ' + rec.get(USERS.LAST_NAME);
+      int lastActivityWeeks = rec.get(STEWARDSHIP.PERFORMED_ON) != null
+          ? (int) ChronoUnit.WEEKS.between(rec.get(STEWARDSHIP.PERFORMED_ON).toLocalDate(), LocalDate.now())
+          : -1;
```

Comment:
💭 Can we change the -1 here to `null`? We'll have to change `lastActivityWeeks`'s type to `Integer` but we won't need to check `if (lastActivityWeeks >= 0)` on line 25 of `api/src/main/java/com/codeforcommunity/dto/site/FilterSitesResponse.java`